frontend/app/src/composables/history/events/tx/use-refresh-transactions.ts

```diff
@@ -367,6 +338,25 @@ export function useRefreshTransactions(): UseRefreshTransactionsReturn {
       logger.error(error);
       resetStatus();
     }
+    finally {
+      finishRefresh();
+
+      // After refresh is complete, check if there are pending accounts to refresh
+      if (get(hasPendingAccounts)) {
+        logger.info('Processing pending accounts after refresh completion');
+        const pendingAccounts = getPendingAccountsForRefresh();
+        // Recursively call refreshTransactions to handle pending accounts
+        setTimeout(() => {
+          refreshTransactions({
+            ...params,
+            payload: {
+              ...params.payload,
+              accounts: pendingAccounts,
+            },
+          }).catch(error => logger.error('Failed to refresh pending accounts', error));
+        }, 100);
+      }
+    }
```

Comment:
small nit: return early her if it does not have pending accounts to reduce the nesting